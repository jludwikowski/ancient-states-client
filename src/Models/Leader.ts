import {Entity} from "./Entity";

export default interface Leader extends Entity {
    leadership: number,
    wisdom: number,
    cunning: number,
    willpower: number,
    imageUrl: ""
}
