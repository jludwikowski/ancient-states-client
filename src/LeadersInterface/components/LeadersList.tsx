import React from "react";
import {LeaderCard} from "./LeaderCard";
import Leader from "../../Models/Leader";

type Props = {
    leadersList: Leader[],
    onDismiss(id: number): void;
};

export const LeadersList: React.FunctionComponent<Props> = ({leadersList, onDismiss}) => (
    <div className="card-group">
        {leadersList && leadersList.length !== 0 ? leadersList.map((leader, index) => (
            <React.Fragment key={leader.id}>
                <LeaderCard leader={leader} onDismiss={onDismiss}/>
            </React.Fragment>
        )) : <div>No Leaders </div>}
    </div>
);
