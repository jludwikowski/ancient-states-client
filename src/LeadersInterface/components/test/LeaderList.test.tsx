import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import Leader from "../../../Models/Leader";
import { LeadersList } from '../LeadersList';

describe('LeadersList', () => {

    const leaders:Leader[] = [{
        name: 'Alcibiades',
        description: 'All fall for Alcibiades',
        leadership: 100,
        wisdom: 0,
        cunning: 100,
        willpower: 0,
        imageUrl: '',
    }];

    const onDismiss = (id:number) => {
        console.log('On onDismiss');
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<LeadersList leadersList={leaders} onDismiss={onDismiss} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Should match snapshot', () => {
        const component = shallow(<LeadersList leadersList={leaders} onDismiss={onDismiss} />);
        expect(component).toMatchSnapshot();
    });
});
