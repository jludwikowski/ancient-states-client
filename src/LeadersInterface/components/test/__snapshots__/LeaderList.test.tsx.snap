// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeadersList Should match snapshot 1`] = `
<div
  className="card-group"
>
  <LeaderCard
    leader={
      Object {
        "cunning": 100,
        "description": "All fall for Alcibiades",
        "imageUrl": "",
        "leadership": 100,
        "name": "Alcibiades",
        "willpower": 0,
        "wisdom": 0,
      }
    }
    onDismiss={[Function]}
  />
</div>
`;
